dashboard/src/utils/colorUtils.ts

```diff
@@ -0,0 +1,86 @@
+const colorCache = new Map<string, string>();
+
+let colorTestElement: HTMLDivElement | null = null;
+
+function getColorTestElement(): HTMLDivElement {
+  if (!colorTestElement) {
+    colorTestElement = document.createElement('div');
+    colorTestElement.style.display = 'none';
+    document.body.appendChild(colorTestElement);
+  }
+  return colorTestElement;
+}
+
+/**
+ * Clear the color cache
+ */
+export function clearColorCache(): void {
+  colorCache.clear();
+}
+
+/**
+ * Get a CSS variable's computed color and convert OKLCH → hex.
+ * Works for both direct color strings and CSS variable references.
+ * @param variableName - The CSS variable name (e.g., '--my-color') or direct color string (e.g., '#ff0000', 'rgb(255, 0, 0)')
+ * @param element - The HTML element to get the CSS variable from (defaults to document.documentElement)
+ * @returns The resolved hex color string (e.g., '#rrggbbaa'), or the original input if not a CSS-variable.
+ */
+export function getResolvedHexColor(
```

Comment:
So I suggest we add constraint to the colors for tailwind for these. If we only allow hex values, and only allow this function to accept tailwind variables, it can be simplified down to:

```ts
export function getResolvedHexColor(
  variableName: `--${string}`
): string {
  if (typeof window === 'undefined') return variableName;

  return getComputedStyle(document.documentElement).getPropertyValue(variableName)
}
```

So I think it's worth it to add constraints for the simplicity of the solution here 😄